### PR TITLE
Complete GHA Migration Process

### PR DESCRIPTION
**Pull request by mallardduck**
This PR completes the process of rebuilding CI from drone to GHA.

This change:
- moves all CI to GHA,
- replaces some bash scripts with native GHA workflows,
- drops the requirement for dapper usage, and
- updates the docs for changes.

### Blockers:

- [x] Waiting on EIO ticket for settings changes: https://github.com/rancherlabs/eio/issues/2191
- [x] Waiting on EIO ticket for secrets: https://github.com/rancherlabs/eio/issues/2199
- [x] Update the PR to properly use secrets set by EIO

### Workflow Examples:
- [Publish with 1 new image](https://github.com/mallardduck/rancher-kubectl/actions/runs/8802578641)
- [Publish with many new images](https://github.com/mallardduck/rancher-kubectl/actions/runs/8803173462)
- [Branch clean up job](https://github.com/mallardduck/rancher-kubectl/actions/runs/8803318728)

### CODE REVIEW

File: .github/workflows/check-new-versions.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Check out repository code
-        uses: actions/checkout@1d96c772d19495a3b5c517cd2bc0cb401ea0529f # v4.1.3
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```
Comment (joshmeranda) [inline .github/workflows/check-new-versions.yml:15]: Whats the reason for downgrading the the version here?

Comment (mallardduck) [reply .github/workflows/check-new-versions.yml:15]: ohh great catch - p sure that came from https://github.com/rancher/kubectl/pull/131 being merged a sloppy rebase.
will fix RN

Comment (mallardduck) [reply .github/workflows/check-new-versions.yml]: rebased to amend commit to not downgrade. 😄

**Comment by mallardduck** [comment]
I just updated this PR to include https://github.com/rancher/kubectl/pull/134 as well since that PR won't catch the new workflows that also need updating.

**Review by joshmeranda — APPROVED** [review]

**Pull request merged**